src/jdbc/cubrid/jdbc/driver/CUBRIDBlob.java

```diff
@@ -124,18 +127,23 @@ public long length() throws SQLException {
 
 		byte[] buf = new byte[length];
 
-		while (length > 0) {
-			read_len = Math.min(length, BLOB_MAX_IO_LENGTH);
-			real_read_len = conn.lobRead(lobHandle.getPackedLobHandle(), pos,
-					buf, total_read_len, read_len);
+		if (isLobLocator) {
+			while (length > 0) {
+				read_len = Math.min(length, BLOB_MAX_IO_LENGTH);
+				real_read_len = conn.lobRead(lobHandle.getPackedLobHandle(), pos,
+						buf, total_read_len, read_len);
 
-			pos += real_read_len;
-			length -= real_read_len;
-			total_read_len += real_read_len;
+				pos += real_read_len;
+				length -= real_read_len;
+				total_read_len += real_read_len;
 
-			if (real_read_len == 0) {
-				break;
+				if (real_read_len == 0) {
+					break;
+				}
 			}
+		} else {
+			System.arraycopy(lobHandle.getPackedLobHandle(), (int) pos, buf, 0, length);
+			total_read_len = length;
```

Comment:
don't need to handle the exception for length = 0 ?

Reply:
The 'if (length <= 0)' is a few lines above.